src/ConfigWidgets/SymbolsDialog.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ConfigWidgets/SymbolsDialog.h"
+
+#include <qpushbutton.h>
+
+#include <QFileDialog>
+#include <QPushButton>
+#include <algorithm>
+#include <memory>
+
+#include "OrbitBase/Logging.h"
+#include "ui_SymbolsDialog.h"
+
+namespace orbit_config_widgets {
+
+SymbolsDialog::~SymbolsDialog() noexcept = default;
+
+SymbolsDialog::SymbolsDialog(QWidget* parent)
+    : QDialog(parent), ui_(std::make_unique<Ui::SymbolsDialog>()) {
+  ui_->setupUi(this);
+
+  QObject::connect(ui_->addButton, &QPushButton::clicked, this, [this]() {
```

Comment:
Is the 3rd parameter really needed when you connect to a lambda? Bit nitpicky, but I'd prefer to have this method as a proper slot instead of a lambda, since `this` is already a `QObject`.

In this case you could also connect it in the designer directly, but I'm not sure if this is something we want to do in general since it's less explicit than in code.

Reply:
We learned the hard way that signals do not respect threads when landing via connection established without the third argument.

Reply:
Interesting, I thought that the overload without context would then call the lambda in the event loop of the main thread, but this doesn't seem to be true. Thanks for pointing that out. On top, the context makes sure to bind the connection to the lifetime of the object, which I also did not think of.

Reply:
I think the discussion about the 3 vs 4 argument overloads is already finished. Thanks Dimitry for chiming in. For all the reasons mentioned I think we should never use the 3 argument overload. 

> In this case you could also connect it in the designer directly, but I'm not sure if this is something we want to do in general since it's less explicit than in code.

I do not know (yet) how to do this in designer and cant really judge which way is more elegant. Since we are doing most connection in code, I am going to leave it as is for now. (I might make this lambda a separate function with dedup handling and tests)